test/distributed/org/apache/cassandra/distributed/upgrade/ConsistencyTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.upgrade;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.UpgradeableCluster;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.apache.cassandra.distributed.api.Feature;
+import org.apache.cassandra.distributed.api.ICoordinator;
+import org.apache.cassandra.distributed.api.IUpgradeableInstance;
+import org.apache.cassandra.distributed.shared.Versions;
+
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.*;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+public class ConsistencyTest extends UpgradeTestBase
+{
+    @Test
+    public void testConsistency() throws Throwable
+    {
+        testConsistency(allUpgrades(3, 1));
```

Comment:
Trying to run this is generating some noise in the logs and OOMs for me...

```
ERROR 21:27:38 LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@182e3d57) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@384421922:Memory@[7f9064b42190..7f9064b424b0) was not released before the reference was garbage collected
ERROR [Reference-Reaper] node1 2020-12-08 15:27:38,081 Ref.java:228 - LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@182e3d57) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@384421922:Memory@[7f9064b42190..7f9064b424b0) was not released before the reference was garbage collected
ERROR 21:27:45 LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@12d65627) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@1473567257:Memory@[7f9057a46640..7f9057a46644) was not released before the reference was garbage collected
ERROR [Reference-Reaper] node1 2020-12-08 15:27:45,696 Ref.java:228 - LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@12d65627) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@1473567257:Memory@[7f9057a46640..7f9057a46644) was not released before the reference was garbage collected
ERROR 21:27:47 LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@3ef99131) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@1724460556:Memory@[7f9057abde30..7f9057abde78) was not released before the reference was garbage collected
ERROR [Reference-Reaper] node1 2020-12-08 15:27:47,828 Ref.java:228 - LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@3ef99131) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@1724460556:Memory@[7f9057abde30..7f9057abde78) was not released before the reference was garbage collected
INFO  [node1_MemtableFlushWriter:1] node1 2020-12-08 15:27:53,538 LogTransaction.java:240 - Unfinished transaction log, deleting /private/var/folders/4d/zfjs7m7s6x5_l93k33r5k6680000gn/T/dtests8093906131697840956/node1/data2/system/compaction_history-b4dbb7b4dc493fb5b3bfce6e434832ca/na-1-big-Data.db 
INFO  [node1_MemtableFlushWriter:1] node1 2020-12-08 15:27:54,042 LogTransaction.java:240 - Unfinished transaction log, deleting /private/var/folders/4d/zfjs7m7s6x5_l93k33r5k6680000gn/T/dtests8093906131697840956/node1/data2/system/compaction_history-b4dbb7b4dc493fb5b3bfce6e434832ca/na-1-big-Index.db 
INFO  [node1_MemtableFlushWriter:1] node1 2020-12-08 15:27:56,281 LogTransaction.java:240 - Unfinished transaction log, deleting /private/var/folders/4d/zfjs7m7s6x5_l93k33r5k6680000gn/T/dtests8093906131697840956/node1/data2/system/compaction_history-b4dbb7b4dc493fb5b3bfce6e434832ca/na_txn_flush_2f723f70-399c-11eb-a318-6dfe4db8d7d4.log 
ERROR 21:28:09 LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@460e3b9c) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@939205974:Memory@[7f91114aab70..7f91114aab74) was not released before the reference was garbage collected
ERROR 21:28:03 Exception in thread Thread[node1_MemtableFlushWriter:1,5,node1]
java.lang.OutOfMemoryError: Metaspace
ERROR [Reference-Reaper] node1 2020-12-08 15:28:09,243 Ref.java:228 - LEAK DETECTED: a reference (org.apache.cassandra.utils.concurrent.Ref$State@460e3b9c) to class org.apache.cassandra.io.util.SafeMemory$MemoryTidy@939205974:Memory@[7f91114aab70..7f91114aab74) was not released before the reference was garbage collected
ERROR [node1_isolatedExecutor:2] node1 2020-12-08 15:28:03,445 CassandraDaemon.java:492 - Exception in thread Thread[node1_MemtableFlushWriter:1,5,node1]
java.lang.OutOfMemoryError: Metaspace
```

Reply:
@adelapena We might just have to separate the tests by starting version, etc.

Reply:
Do you mean a method or a class per starting version?

Reply:
Unless there's a way we can clear metaspace between methods, it might have to be separate classes?

Reply:
Done [here](https://github.com/apache/cassandra/pull/801/commits/4e414b982534e31e549fba80d86fbb290b644bbe).